CONTRIBUTING.md

```diff
@@ -0,0 +1,388 @@
+# Contributing
+
+## Getting Started
+
+1. Fork the repository on GitHub
+2. Clone the forked repository to your local machine
+3. Install the dependencies of the frontend with `npm ci`
+4. Install the dependencies of the backend with `pip install -r requirements.txt`
+
+Chainner is an Electron application that uses Python for the backend. The frontend and backend communicate through a JSON API. The frontend is located in the `src` folder and the backend is located in the `backend` folder.
+
+We use VSCode as our IDE of choice, and we strongly recommend you to do the same. We have included the `.vscode` folder in this repository, which contains some settings that make developing chainner easier.
+
+## Running the Application
+
+There are 2 way to run chainner:
+
+1. Use `npm run dev` to run the application in development mode. This will run the frontend and backend in parallel. The frontend will be reloaded automatically when you make changes to the code. The backend will be reloaded automatically when you make changes to the code and restart the frontend.
+
+    Note: The backend will use your system python in this mode. This means that you might need to (re)install some dependencies using chainner's dependency manager.
+
+1. Use `npm start` to run the application in production mode. The frontend will automatically start and manage the backend process. The frontend will be reloaded automatically when you make changes to the code. The backend will **NOT** be reloaded automatically, use `npm run dev` if you need this.
+
+    NOTE: Chainner will use its own python installation, the same that a regular installation of chainner also uses.
+
+Generally, use `npm run dev` when you want to develop the backend or the frontend, and use `npm start` you want to start chainner in a more production-like mode.
+
+## Project structure
+
+The project is split into 2 parts: the frontend and the backend. The frontend is located in the `src` folder and the backend is located in the `backend` folder.
+
+### Frontend
+
+The frontend is an Electron application written in TypeScript. As such, the frontend is split into 3 parts:
+
+1. The code for the main Node.js electron process, which is located in `src/main`.
+2. The code for the renderer process, which is located in `src/renderer`. This is the web part of the application. We use React and Chakra UI for the frontend.
+3. The code for the shared code between the main and renderer process, which is located in `src/common`.
+
+### Backend
+
+The backend is a Python application. The backend is roughly split into 3 parts:
+
+1. `backend/src/packages` contains the packages that contain chainner's nodes. Most nodes are in the `chaiNNer_standard` package.
+2. `backend/src/nodes` contains logic shared between nodes. This includes common functionality for task such as upscaling, classes to define inputs and outputs, etc.
```

Comment:
I might be changing this soon so this is just a note to myself to remember to update this doc if i end up doing that